CourseSystem/BLL/Services/GroupService.cs

```diff
@@ -254,4 +276,36 @@ private async Task<Result<bool>> SetStudentProgressInUserGroup(Group group, AppU
     
         return new Result<bool>(true);
     }
+    
+    private async Task<List<Group>> CheckStartAndEndGroupDate(List<Group> groups)
+    {
+        foreach (var group in groups)
+        {
+            SetGroupStatus(group);
+
+            await _repository.UpdateAsync(group);
+        }
+
+        await _unitOfWork.Save();
+
+        return groups;
+    }
+    
+    private void SetGroupStatus(Group group)
```

Comment:
Just for you to know, in future, group and assignment status will be updated separately in special helper that will also add notifications.